### PR TITLE
User types and initial permission checks for them

### PR DESCRIPTION
**Pull request by tuomas777**
There are now two types of users, an official and a contractor. For setting a user's type there a flags `is_official` and `is_contractor` in the User model.

Non-authenticated users can no longer do anything else than just POST new events. Contractors can view and modify events in their own contract zones. Officials have no restrictions atm.

Added an initial model for contract zones to make it possible to implement the contract zone based restrictions. We don't have data for those zones yet, so the model is likely to change quite a bit.

Enabled also Tunnistamo based API token authentication.

### CODE REVIEW

File: areas/models.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+from django.contrib.gis.db import models
+from django.utils.translation import ugettext_lazy as _
+from munigeo.utils import get_default_srid
+
+PROJECTION_SRID = get_default_srid()
+
+
+class ContractZone(models.Model):
+    name = models.CharField(verbose_name=_('name'), max_length=255)
```
Comment (igordavydsson) [inline areas/models.py]: Do we need translations for those or no?

File: users/models.py [hunk 2]
```diff
@@ -8,4 +8,6 @@
 class User(AbstractUser):
     is_official = models.BooleanField(verbose_name=_('official'), default=False)
     is_contractor = models.BooleanField(verbose_name=_('contractor'), default=False)
-
+    contract_zones = models.ManyToManyField(
```
Comment (igordavydsson) [inline users/models.py]: Can one zone have multiple contractors?

File: areas/models.py (hunk 1 above)
Comment (tuomas777) [reply areas/models.py]: No idea, so I opted for the simpler solution for now :)

File: users/models.py (hunk 2 above)
Comment (tuomas777) [reply users/models.py]: Yes I think it should be possible.

**Pull request merged**